pkg/registry/core/service/strategy_test.go

```diff
@@ -812,6 +888,31 @@ func TestDropTypeDependentFields(t *testing.T) {
 			svc:    makeValidServiceCustom(setTypeLoadBalancer, setAllocateLoadBalancerNodePortsFalse),
 			patch:  patches(setTypeNodePort, setAllocateLoadBalancerNodePortsTrue),
 			expect: makeValidServiceCustom(setTypeNodePort, setAllocateLoadBalancerNodePortsTrue),
+		}, { // loadBalancerClass cases
+			name:   "clear loadBalancerClass when changed service type",
+			svc:    makeValidServiceCustom(setTypeLoadBalancer, setLoadBalancerClass),
+			patch:  setTypeClusterIP,
+			expect: makeValidServiceCustom(setTypeClusterIP, clearLoadBalancerClass),
+		}, {
+			name:   "keep original loadBalancerClass when changed load balancer class name",
+			svc:    makeValidServiceCustom(setTypeLoadBalancer, setLoadBalancerClass),
+			patch:  changeLoadBalancerClass,
+			expect: makeValidServiceCustom(setTypeClusterIP, setLoadBalancerClass),
+		}, {
+			name:   "keep original loadBalancerClass when clear load balancer class name",
+			svc:    makeValidServiceCustom(setTypeLoadBalancer, setLoadBalancerClass),
+			patch:  clearLoadBalancerClass,
+			expect: makeValidServiceCustom(setTypeClusterIP, setLoadBalancerClass),
+		}, {
+			name:   "clear loadBalancerClass when changed service type and load balancer class",
+			svc:    makeValidServiceCustom(setTypeLoadBalancer, setLoadBalancerClass),
+			patch:  patches(setTypeClusterIP, changeLoadBalancerClass),
+			expect: makeValidServiceCustom(setTypeClusterIP, clearLoadBalancerClass),
+		}, {
+			name:   "set loadBalancerClass when changed service type to load balancer",
+			svc:    makeValidServiceCustom(setTypeClusterIP),
+			patch:  patches(setTypeLoadBalancer, setLoadBalancerClass),
+			expect: makeValidServiceCustom(setTypeLoadBalancer, setLoadBalancerClass),
 		}}
 
 	for _, tc := range testCases {
```

Comment:
You didn't add anything to check the value of this field in the test itself!

Reply:
My bad, forgot to check how the test cases get checked.